File: xa-api-python-sdk/xa/model/Action.py
from xa.model.Operation import Operation


class Action(Operation):

    def __init__(self, eventName, value=0, timestamp=None):
        self.eventName = eventName
        self.value = value
        self.timestamp = timestamp

    def getOperationName(self, index):
        return 'action' + str(index)

    def getOperationValue(self):
        value = self.eventName + ',' + str(self.value)
        if None != self.timestamp:
            value += ',' + str(self.timestamp)
        return value

File: xa-api-php-sdk/xa/model/Action.php
<?php
include_once '../../model/Operation.php' ;

class Action implements \Operation {
	
	var $eventName;
	var $value;
	var $timestamp;
	
	
	public function __construct($eventName,$value = null,$timestamp = null){
		$this->eventName = $eventName ;
		$this->value = $value;
		$this->timestamp = $timestamp;
	}
	
	public function getOperationValue() {
		return $this->eventName . ',' . $this->value . ',' . $this->timestamp ;
	}
	
	public function getOperationName($index) {
		return 'action' . $index ;
	}
}

?>